pkg/vm/createEvm.go

```diff
@@ -5,43 +5,48 @@ package vm
 import (
 	"bytes"
 	"encoding/json"
-	"errors"
 	"fmt"
 	"math/big"
 
-	"github.com/ava-labs/avalanche-cli/cmd/prompts"
 	"github.com/ava-labs/avalanche-cli/ux"
 	"github.com/ava-labs/avalanchego/utils/logging"
 	"github.com/ava-labs/subnet-evm/core"
 	"github.com/ava-labs/subnet-evm/params"
 )
 
-func CreateEvmGenesis(name string, log logging.Logger) ([]byte, error) {
+func CreateEvmGenesis(name string, log logging.Logger) ([]byte, string, error) {
 	ux.Logger.PrintToUser("creating subnet %s", name)
 
 	genesis := core.Genesis{}
 	conf := params.SubnetEVMDefaultChainConfig
 
-	chainId, err := getChainId()
-	if err != nil {
-		return []byte{}, err
-	}
-	conf.ChainID = chainId
-
-	*conf, err = getFeeConfig(*conf)
-	if err != nil {
-		return []byte{}, err
-	}
-
-	allocation, err := getAllocation()
-	if err != nil {
-		return []byte{}, err
+	stage := startStage
+
+	var chainId *big.Int
+	var tokenName string
+	var allocation core.GenesisAlloc
+	var err error
+
+	for stage != doneStage {
+		switch stage {
+		case startStage:
+			stage = descriptorStage
+		case descriptorStage:
+			chainId, tokenName, stage, err = getDescriptors()
+			fmt.Println("Creating token", tokenName)
```

Comment:
Is this a case for `ux.Logger.PrintToUser` ?
Also, I usually think that it's a good idea to log each stage change to the logs.

It might be a bit overkill here so leaving it to your judgment. But recommended.